plugins/sim/cdsimcontroller.cpp

```diff
@@ -402,28 +491,45 @@ void CDSimController::ensureSimContactsPresent()
 
 void CDSimController::voicemailConfigurationChanged()
 {
-    if (!m_voicemailConf || !m_simManager.present()) {
+    ModemData *modem = 0;
+    if (MGConfItem *voicemailConf = qobject_cast<MGConfItem *>(sender())) {
```

Comment:
Actually, it's possible that if I move some of these slots and private functions into the ModemData class, then they will be effectively unchanged, apart from the class name.  That might be the best way to achieve a minimal diff.  I'll test that change and compare.